src/main/java/com/mcmoddev/mmdbot/modules/commands/community/CmdEvaluate.java

```diff
@@ -0,0 +1,239 @@
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ * https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ */
+package com.mcmoddev.mmdbot.modules.commands.community;
+
+import static com.mcmoddev.mmdbot.utilities.scripting.ScriptingUtils.ALLOWED_MENTIONS;
+import static com.mcmoddev.mmdbot.utilities.scripting.ScriptingUtils.createGuild;
+import static com.mcmoddev.mmdbot.utilities.scripting.ScriptingUtils.createMember;
+import static com.mcmoddev.mmdbot.utilities.scripting.ScriptingUtils.createMessageChannel;
+import static com.mcmoddev.mmdbot.utilities.scripting.ScriptingUtils.createTextChannel;
+import static com.mcmoddev.mmdbot.utilities.scripting.ScriptingUtils.createUser;
+import static com.mcmoddev.mmdbot.utilities.scripting.ScriptingUtils.getEmbedFromValue;
+import static com.mcmoddev.mmdbot.utilities.scripting.ScriptingUtils.validateArgs;
+import com.jagrosh.jdautilities.command.CommandEvent;
+import com.jagrosh.jdautilities.command.SlashCommand;
+import com.jagrosh.jdautilities.command.SlashCommandEvent;
+import com.mcmoddev.mmdbot.modules.commands.DismissListener;
+import com.mcmoddev.mmdbot.utilities.Utils;
+import com.mcmoddev.mmdbot.utilities.scripting.ScriptingContext;
+import com.mcmoddev.mmdbot.utilities.scripting.ScriptingUtils;
+import net.dv8tion.jda.api.MessageBuilder;
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.MessageChannel;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Objects;
+
+public class CmdEvaluate extends SlashCommand {
+
+    public CmdEvaluate() {
+        guildOnly = true;
+        name = "evaluate";
+        aliases = new String[] {"eval"};
+        help = "Evaluates the given script";
+        options = List.of(new OptionData(OptionType.STRING, "script", "The script to evaluate.").setRequired(true));
+    }
+
+    @Override
+    protected void execute(final SlashCommandEvent event) {
+        event.deferReply().queue(hook -> {
+            try {
+                ScriptingUtils.evaluate(Utils.getOrEmpty(event, "script"), createContext(new EvaluationContext() {
```

Comment:
Would it not be a good idea to cache these contexts? Creating a new one and discarding it every time a slash command is invoked seems like a not-good idea.

Reply:
That is not possible, due to the fact that each context has "bindings" (which contain all of the context members: functions, classes, variables), and those bindings cannot be discarded.

With that being said, contexts are designed to be created for every evaluation. And such, the first evaluation may take 2~3 seconds, but after that the instantiation is almost instant.